test/components/guideLineLayerTests.ts

```diff
@@ -0,0 +1,401 @@
+    let timeScaleGLL = new Plottable.Components.GuideLineLayer<Date>("vertical");
+    let timeScale = new Plottable.Scales.Time();
+    timeScaleGLL.scale(timeScale);
+    assert.doesNotThrow(() => timeScaleGLL.destroy(), Error, "destroy() does not error if a scale was set");
+    assert.strictEqual((<any>timeScale)._callbacks.size, 0, "callback was removed from Scale");
+  });
+
+  describe("coordination between scale(), value(), and pixelPosition()", () => {
+    it("changing value() updates pixelPosition() if scale() is set", () => {
+      let linearScale = new Plottable.Scales.Linear();
+      linearScale.domain([0, 1]);
+      linearScale.range([0, 100]);
+      let gll = new Plottable.Components.GuideLineLayer<number>("vertical");
+      gll.scale(linearScale);
+      let value = 0.5;
+      let expectedPosition = linearScale.scale(value);
+      gll.value(value);
+      assert.strictEqual(gll.pixelPosition(), expectedPosition, "pixel position was updated to match the set value");
+    });
+
+    it("changing pixelPosition() updates value() if scale() is set", () => {
+      let linearScale = new Plottable.Scales.Linear();
+      linearScale.domain([0, 1]);
+      linearScale.range([0, 100]);
+      let gll = new Plottable.Components.GuideLineLayer<number>("vertical");
+      gll.scale(linearScale);
+      let position = 50;
+      let expectedValue = linearScale.invert(position);
+      gll.pixelPosition(position);
+      assert.strictEqual(gll.value(), expectedValue, "value was updated to match the set pixel position");
+    });
+
+    it("updating the scale's domain updates pixelPosition() to match the existing value()", () => {
+      let linearScale = new Plottable.Scales.Linear();
+      linearScale.domain([0, 1]);
+      linearScale.range([0, 100]);
+      let gll = new Plottable.Components.GuideLineLayer<number>("vertical");
+      gll.scale(linearScale);
+      let value = 0.5;
+      gll.value(value);
+      assert.strictEqual(gll.pixelPosition(), linearScale.scale(value), "pixel position matches set value");
+      linearScale.domain([0, 2]);
+      assert.strictEqual(gll.pixelPosition(), linearScale.scale(value), "pixel position was updated when scale updated");
+    });
+
+    it("changing the scale updates pixelPosition() if value() is set", () => {
+      let linearScale = new Plottable.Scales.Linear();
+      linearScale.domain([0, 1]);
+      linearScale.range([0, 100]);
+
+      let gll = new Plottable.Components.GuideLineLayer<number>("vertical");
+      let value = 0.5;
+      gll.value(value);
+      let setPosition = -999;
+      gll.pixelPosition(setPosition);
+      let expectedPosition = linearScale.scale(value);
+      gll.scale(linearScale);
+      assert.strictEqual(gll.pixelPosition(), expectedPosition, "setting the scale updates the pixel position");
+      assert.notStrictEqual(gll.pixelPosition(), setPosition, "originally-set pixel position was overridden");
```

Comment:
If `expectedPosition === setPosition`, then the test would pass whether or not this functionality worked. It makes sense to error in that case to guard against changes to the code, right?